src/util/codec/datum.rs

```diff
@@ -153,7 +166,7 @@ impl Datum {
 
     fn cmp_f64(&self, ctx: &EvalContext, f: f64) -> Result<Ordering> {
         match *self {
-            Datum::Null | Datum::Min => Ok(Ordering::Less),
+            Datum::Null | Datum::Min | Datum::Json(_) => Ok(Ordering::Less),
```

Comment:
It should not be `Datum::Json` here. But if something is wrong(a bug maybe), the code executes this branch. It should handle properly and silently by converting `f`, or panic and then crash wildly. The former is preferred.